xbmc/video/tags/ISetInfoTagLoader.h

```diff
@@ -0,0 +1,61 @@
+/*
+ *  Copyright (C) 2017-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "InfoScanner.h"
+#include "addons/Scraper.h"
+
+#include <string>
+#include <utility>
+
+class CFileItem;
+class CSetInfoTag;
+class EmbeddedArt;
+
+namespace VIDEO
+{
+
+//! \brief Base class for set tag loaders.
+class ISetInfoTagLoader
+{
+public:
+  //! \brief Constructor
+  //! \param item The item to load info for
+  //! \param info Scraper info
+  //! \param llokInFolder True to look in folder holding file
+  ISetInfoTagLoader(const CFileItem& item, ADDON::ScraperPtr info, bool lookInFolder = false)
+    : m_item(item), m_info(std::move(info))
+  {
+  }
+  virtual ~ISetInfoTagLoader() = default;
+
+  //! \brief Returns true if we have info to provide.
+  virtual bool HasInfo() const = 0;
+
+  //! \brief Load tag from file.
+  //! \brief tag Tag to load info into
+  //! \brief prioritise True to prioritise data over existing data in tag
+  //! \returns True if tag was read, false otherwise
+  virtual CInfoScanner::INFO_TYPE Load(CSetInfoTag& tag,
+                                       bool prioritise,
+                                       std::vector<EmbeddedArt>* art = nullptr) = 0;
+
+  //! \brief Returns url associated with obtained URL (NFO_URL et al).
+  const CScraperUrl& ScraperUrl() const { return m_url; }
+
+  //! \brief Returns current scaper info.
+  const ADDON::ScraperPtr GetAddonInfo() const { return m_info; }
+
+protected:
```

Comment:
Any reason this is not `private`? Do you expect other classes to be derived from this one? I don't think so.
I have seen this at other places in this PR. maybe you rethink this everywhere you introduced a new class.

Reply:
Again, just copied from `IVideoInfoTagLoader`. Will try with it as private.

Reply:
It doesn't work. `SetTagLoaderNFO` seems to be derived (in some way) from it.

Reply:
Please let the person who opened the request to change something resolve, okay? Otherwise it is really hard to follow what was realised and what not.

Reply:
> Do you expect other classes to be derived from this one?

Answering my own question. Yes! This is an abstract base class supposed to have subclasses. So, `protected` is fine here.